third_party/move/move-compiler-v2/src/pipeline/control_flow_graph_simplifier.rs

```diff
@@ -469,6 +505,80 @@ impl ControlFlowGraphCodeGenerator {
                         .collect::<BTreeSet<_>>()
             );
         }
+        // check invariant 3
+        for (block, code) in self.code_blocks.iter() {
+            if self.is_trivial_block(*block) {
+                assert!(code.is_empty())
+            } else {
+                assert!(!code.is_empty(), "block {} is empty", block);
+            }
+        }
+        // check invariant 4
+        assert!(self.get_the_successor(self.entry_block) != self.exit_block);
+        // check invariant 5
+        for (block, succs) in self.successors.iter() {
+            if block != &self.exit_block {
+                assert!(!succs.is_empty(), "block {} has no successors", block);
+            }
+        }
+        // check invariant 6
```

Comment:
Invariant 6 says: "All non-trivial blocks with non-trivial predecessors must start with a label."

The code here however is not doing this check. In particular, if we have a non-trivial successor of the entry block that also has other non-trivial predecessors, then that block must begin with a label: this is not checked by the code here.

Reply:
done